bids/layout/models.py

```diff
@@ -717,3 +723,45 @@ class FileAssociation(Base):
                              Column('config', String, ForeignKey('configs.name')),
                              Column('entity', String, ForeignKey('entities.name'))
                              )
+
+
+class DerivativeDatasets(UserDict):
+    def __getitem__(self, key):
+        try:
+            return super().__getitem__(key)
+        except KeyError:
+            pass
+
+        try:
+            result = self.get_pipeline(key)
+            print(
+                "[DEPRECATION WARNING] Directly selecting derivative datasets using "
+                "pipeline name (i.e. dataset.derivatives[<pipeline_name>] will be "
+                "phased out in an upcoming release. Select instead using the folder "
+                "name of the dataset (i.e. dataset.derivatives[<folder_name>]), or use "
+                "dataset.derivatives.get_pipeline(<pipeline_name>)."
+            )
```

Comment:
This should be a deprecation warning.

```suggestion
            warnings.warn(
                "Directly selecting derivative datasets using pipeline name "
                "(i.e., dataset.derivatives[<pipeline_name>] will be phase out in "
                "an upcoming release. Select instead using the folder name of the "
                "dataset (i.e., dataset.derivatives[<folder_name>]), or use "
                "dataset.derivatives.get_pipeline(<pipeline_name>).",
                DeprecationWarning
            )
```